flake.nix

```diff
@@ -0,0 +1,74 @@
+{
+  description = "Simon’s Improved Layout Engine";
+
+  # To make user overrides of the nixpkgs flake not take effect
+  inputs.nixpkgs.url = "github:nixos/nixpkgs/nixos-unstable";
+  inputs.flake-utils.url = "github:numtide/flake-utils";
+
+  # https://nixos.wiki/wiki/Flakes#Using_flakes_project_from_a_legacy_Nix
+  inputs.flake-compat = {
+    url = "github:edolstra/flake-compat";
+    flake = false;
+  };
+
+  outputs = { self
+    , nixpkgs
+    , flake-utils
+    , flake-compat
+  }:
+  flake-utils.lib.eachDefaultSystem (system:
+    let
+      pkgs = import nixpkgs {
+        inherit system;
+      };
+      # TODO: Should this be replaced with libtexpdf package from nixpkgs? or
+      # should we keep it that way, so that it'd be easy to test new versions
+      # of libtexpdf when developing?
+      libtexpdf-src = builtins.fetchGit {
+        url = "https://github.com/sile-typesetter/libtexpdf";
+        rev = "${(pkgs.lib.fileContents "${self}/libtexpdf.git-rev")}";
+      };
+      # Use the expression from Nixpkgs instead of rewriting it here.
+      sile = pkgs.sile.overrideAttrs(oldAttr: rec {
+        version = "${(pkgs.lib.importJSON ./package.json).version}-flake";
```

Comment:
This is a nice touch and definitely better than not having it at all. However I'm wondering if we can do one better. Do you have access to the actual commit SHA at this point? Normally when building from Git, SILE will identify itself with a counter and short SHA of the sources it was built from, e.g.:

```console
$ SILE v0.11.1.r144-g8935c52 (Lua 5.4)
```

Obviously this is based on `git describe` output, but I think those are useful bits to know.

Not a big deal if there isn't an obvious way to do it, but since I'm planning on using this Flake to build the SILE website it would probably be useful to see from CI which version was used to generate examples.

Reply:
That wasn't trivial, because `.git` is not available inside the source during build. I did found a way though. I pushed a fixup commit for this.

Reply:
BTW would you like to add packages of sile built with different `lua` versions? Currently Nixpkgs defaults to lua 5.2

Reply:
Would be cool indeed; the current `SILE v0.11.1-flake (Lua 5.2)` is kinda less than ideal in that it doesn't show it's not *really* 0.11.1, there are quite some commits over it already. But for me, I'm happy this PR is making my life easier, even if imperfect, so I personally (FWIW) see it as something that could be perfected post-merge.

With that said, I seem to vaguely recall there were in theory some extra "magic" attributes that are maybe available in a flake expression with some extra info; though I wasn't able to google them up yet. One thing I found is ["Flake reference attributes"](https://github.com/NixOS/nix/blob/master/src/nix/flake.md#flake-reference-attributes), but I didn't manage to access them; maybe those are only for inputs? Dunno. I always have trouble finding stuff in Nix docs *\<sigh>*

Reply:
Re Lua versions: https://github.com/NixOS/nixpkgs/issues/138420